diff --git a/bluesky/utils/__init__.py b/bluesky/utils/__init__.py
index ae474e2d..12b76cc5 100644
--- a/bluesky/utils/__init__.py
+++ b/bluesky/utils/__init__.py
@@ -610,10 +610,10 @@ def snake_cyclers(cyclers, snake_booleans):
     new_cyclers = []
     for c in cyclers:
         lengths.append(len(c))
-    total_length = np.product(lengths)
+    total_length = np.prod(lengths)
     for i, (c, snake) in enumerate(zip(cyclers, snake_booleans)):
-        num_tiles = np.product(lengths[:i])
-        num_repeats = np.product(lengths[i+1:])
+        num_tiles = np.prod(lengths[:i])
+        num_repeats = np.prod(lengths[i+1:])
         for k, v in c._transpose().items():
             if snake:
                 v = v + v[::-1]
