diff --git a/caproto/_backend.py b/caproto/_backend.py
index ab91f158..ab5fea3b 100644
--- a/caproto/_backend.py
+++ b/caproto/_backend.py
@@ -337,11 +337,9 @@ def convert_values(values, from_dtype, to_dtype, *, direction,
             return _encode_to_string_array(values, string_encoding)
         return _decode_string_list(values, string_encoding)
     if to_dtype == ChannelType.CHAR:
-        if len(values):
-            if string_encoding and isinstance(values[0], str):
-                return values
-            if not string_encoding and isinstance(values[0], bytes):
-                return values
+        if isinstance(values[0] if len(values) else values,
+                      str if string_encoding else bytes):
+            return values
 
     byteswap = (auto_byteswap and direction == ConversionDirection.TO_WIRE)
     return backend.python_to_epics(to_dtype, values, byteswap=byteswap,
diff --git a/caproto/server/common.py b/caproto/server/common.py
index a013ed38..efbf0c3e 100644
--- a/caproto/server/common.py
+++ b/caproto/server/common.py
@@ -1193,7 +1193,7 @@ class Context:
         cavc = ca.VirtualCircuit(ca.SERVER, addr, None)
         circuit = self.CircuitClass(cavc, client, self)
         self.circuits.add(circuit)
-        self.log.info('Connected to new client at %s:%d (total: %d).', *addr,
+        self.log.debug('Connected to new client at %s:%d (total: %d).', *addr,
                       len(self.circuits))
 
         await circuit.run()
@@ -1215,7 +1215,7 @@ class Context:
         except KeyboardInterrupt as ex:
             self.log.debug('TCP handler received KeyboardInterrupt')
             raise self.ServerExit() from ex
-        self.log.info('Disconnected from client at %s:%d (total: %d).', *addr,
+        self.log.debug('Disconnected from client at %s:%d (total: %d).', *addr,
                       len(self.circuits))
 
     def stop(self):
