diff -ur ADSupport-R1-10/supportApp/GraphicsMagickSrc/jp2/src/libjasper/Makefile ADSupport-R1-10/supportApp/GraphicsMagickSrc/jp2/src/libjasper/Makefile
--- ADSupport-R1-10/supportApp/GraphicsMagickSrc/jp2/src/libjasper/Makefile	2021-05-27 01:17:55.000000000 +0800
+++ ADSupport-R1-10/supportApp/GraphicsMagickSrc/jp2/src/libjasper/Makefile	2025-11-11 03:57:13.484323662 +0800
@@ -5,6 +5,11 @@
   ifeq ($(GRAPHICSMAGICK_EXTERNAL),NO)
 
     LIBRARY = jp2
+    USR_CFLAGS_Linux += -fpermissive
+
+    ifeq (mingw, $(findstring mingw, $(T_A)))
+      USR_CFLAGS_WIN32 += -fpermissive
+    endif
     
     ifeq (windows-x64, $(findstring windows-x64, $(T_A)))
       USR_CFLAGS_WIN32 += -DWIN64
diff -ur ADSupport-R1-10/supportApp/GraphicsMagickSrc/Magick++/lib/Image.cpp ADSupport-R1-10/supportApp/GraphicsMagickSrc/Magick++/lib/Image.cpp
--- ADSupport-R1-10/supportApp/GraphicsMagickSrc/Magick++/lib/Image.cpp	2021-05-27 01:17:55.000000000 +0800
+++ ADSupport-R1-10/supportApp/GraphicsMagickSrc/Magick++/lib/Image.cpp	2023-06-12 23:47:19.295967812 +0800
@@ -16,14 +16,14 @@
 #include <math.h>
 #include <string.h> // for strcpy
 
-using namespace std;
-
 #include "Magick++/Image.h"
 #include "Magick++/Functions.h"
 #include "Magick++/Pixels.h"
 #include "Magick++/Options.h"
 #include "Magick++/ImageRef.h"
 
+using namespace std;
+
 #define _MAGICKLIB_
 #include "../../coders/static.h"
 
diff -ur ADSupport-R1-10/supportApp/GraphicsMagickSrc/Magick++/lib/Makefile ADSupport-R1-10/supportApp/GraphicsMagickSrc/Magick++/lib/Makefile
--- ADSupport-R1-10/supportApp/GraphicsMagickSrc/Magick++/lib/Makefile	2021-05-27 01:17:55.000000000 +0800
+++ ADSupport-R1-10/supportApp/GraphicsMagickSrc/Magick++/lib/Makefile	2023-06-12 23:52:17.165689890 +0800
@@ -5,7 +5,12 @@
   ifeq ($(GRAPHICSMAGICK_EXTERNAL),NO)
 
     LIBRARY = Magick++
+    USR_CXXFLAGS_Linux += -fpermissive
     
+    ifeq (mingw, $(findstring mingw, $(T_A)))
+      USR_CXXFLAGS_WIN32 += -fpermissive
+    endif
+
     ifeq (windows-x64, $(findstring windows-x64, $(T_A)))
       USR_CXXFLAGS_WIN32 += -DWIN64
     endif
diff -ur ADSupport-R1-10/supportApp/hdf5_hlSrc/H5LTanalyze.c ADSupport-R1-10/supportApp/hdf5_hlSrc/H5LTanalyze.c
--- ADSupport-R1-10/supportApp/hdf5_hlSrc/H5LTanalyze.c	2021-05-27 01:17:55.000000000 +0800
+++ ADSupport-R1-10/supportApp/hdf5_hlSrc/H5LTanalyze.c	2023-06-13 00:13:44.053129768 +0800
@@ -62,6 +62,9 @@
 #include <string.h>
 #include <errno.h>
 #include <stdlib.h>
+#ifndef _MSC_VER
+#include <unistd.h>
+#endif
 
 /* end standard C headers. */
 
diff -ur ADSupport-R1-10/supportApp/hdf5Src/os/WIN32/H5pubconf.h ADSupport-R1-10/supportApp/hdf5Src/os/WIN32/H5pubconf.h
--- ADSupport-R1-10/supportApp/hdf5Src/os/WIN32/H5pubconf.h	2021-05-27 01:17:55.000000000 +0800
+++ ADSupport-R1-10/supportApp/hdf5Src/os/WIN32/H5pubconf.h	2023-06-13 00:01:50.241002866 +0800
@@ -16,7 +16,7 @@
 #define H5_HAVE_WIN32_API 1
 
 /* Define if using a Windows compiler (i.e. Visual Studio) */
-#define H5_HAVE_VISUAL_STUDIO 1
+/* #undef H5_HAVE_VISUAL_STUDIO */
 
 /* Define if building universal (internal helper macro) */
 /* #undef H5_AC_APPLE_UNIVERSAL_BUILD */
diff -ur ADSupport-R1-10/supportApp/hdf5Src/Makefile ADSupport-R1-10/supportApp/hdf5Src/Makefile
--- ADSupport-R1-10/supportApp/hdf5Src/Makefile	2021-05-27 01:17:55.000000000 +0800
+++ ADSupport-R1-10/supportApp/hdf5Src/Makefile	2025-11-11 01:43:54.908445080 +0800
@@ -13,6 +13,9 @@
 
     USR_CFLAGS += -D NDEBUG
     USR_CFLAGS_Linux += -std=c99
+    ifneq (mingw, $(findstring mingw, $(T_A)))
+      USR_CFLAGS_WIN32 += -D H5_HAVE_VISUAL_STUDIO
+    endif
     ifeq ($(SHARED_LIBRARIES), YES)
       USR_CFLAGS_WIN32 += -D H5_BUILT_AS_DYNAMIC_LIB 
       USR_CFLAGS_WIN32 += -D hdf5_shared_EXPORTS
diff -ur ADSupport-R1-10/supportApp/netCDFSrc/libdispatch/dutil.c ADSupport-R1-10/supportApp/netCDFSrc/libdispatch/dutil.c
--- ADSupport-R1-10/supportApp/netCDFSrc/libdispatch/dutil.c	2021-05-27 01:17:55.000000000 +0800
+++ ADSupport-R1-10/supportApp/netCDFSrc/libdispatch/dutil.c	2023-06-13 00:16:32.978106517 +0800
@@ -225,7 +225,7 @@
 	_mktemp_s(tmp,sizeof(tmp)-1);
 #else /*!_MSC_VER*/
         mktemp(tmp);
-	tmo[sizeof[tmp]-1] = '\0';
+	tmp[sizeof(tmp)-1] = '\0';
 #endif
 #else /* !HAVE_MKTEMP */
 	/* Need to simulate by using some kind of pseudo-random number */
diff -ur ADSupport-R1-10/supportApp/netCDFSrc/Makefile ADSupport-R1-10/supportApp/netCDFSrc/Makefile
--- ADSupport-R1-10/supportApp/netCDFSrc/Makefile	2021-05-27 01:17:55.000000000 +0800
+++ ADSupport-R1-10/supportApp/netCDFSrc/Makefile	2025-11-11 01:36:34.402779486 +0800
@@ -12,6 +12,9 @@
     LIBRARY_IOC = netCDF
     NETCDF = $(TOP)/supportApp/netCDFSrc
     USR_CFLAGS += -DHAVE_CONFIG_H
+    ifeq (mingw, $(findstring mingw, $(T_A)))
+      USR_CFLAGS_WIN32 += -DHAVE_SSIZE_T=1
+    endif
 
     ifeq ($(SHARED_LIBRARIES),YES)
       USR_CFLAGS_WIN32 += -DDLL_NETCDF
diff -ur ADSupport-R1-10/supportApp/xml2Src/Makefile ADSupport-R1-10/supportApp/xml2Src/Makefile
--- ADSupport-R1-10/supportApp/xml2Src/Makefile	2021-05-27 01:17:55.000000000 +0800
+++ ADSupport-R1-10/supportApp/xml2Src/Makefile	2023-12-29 19:35:31.771010500 +0800
@@ -1,4 +1,8 @@
 TOP=../..
+ifeq (mingw, $(findstring mingw, $(T_A)))
+  USR_CFLAGS_WIN32 += -D_WIN32_WINNT=0x0600
+endif
+
 include $(TOP)/configure/CONFIG
 #----------------------------------------
 #  ADD MACRO DEFINITIONS AFTER THIS LINE
