diff -ur motor/motorApp/MotorSrc/motordrvCom.h motor/motorApp/MotorSrc/motordrvCom.h
--- motor/motorApp/MotorSrc/motordrvCom.h	2021-02-03 02:19:45.000000000 +0800
+++ motor/motorApp/MotorSrc/motordrvCom.h	2023-04-14 19:43:22.491272387 +0800
@@ -45,6 +45,7 @@
 #ifndef	INCmotordrvComh
 #define	INCmotordrvComh 1
 
+#include <shareLib.h>
 #include <callback.h>
 #include <epicsTypes.h>
 #include <epicsEvent.h>
diff -ur motor/motorApp/MotorSrc/motorRecord.cc motor/motorApp/MotorSrc/motorRecord.cc
--- motor/motorApp/MotorSrc/motorRecord.cc	2021-02-03 02:19:45.000000000 +0800
+++ motor/motorApp/MotorSrc/motorRecord.cc	2023-04-14 19:44:19.831113351 +0800
@@ -2410,7 +2410,7 @@
                 }
                 /* IF move is in preferred direction, AND, current position is within backlash range. */
                 else if ((preferred_dir == true) &&
-                         ((use_rel == true  && relbpos <= 1.0) ||
+                         ((use_rel == true  && ((pmr->bdst >= 0 && relbpos >= 1.0) || (pmr->bdst < 0 && relbpos <= -1.0))) ||
                           (use_rel == false && (fabs(newpos - currpos) <= rbdst1))
                          )
                         )
