diff -ur support-R6-2-1/Makefile support-R6-2-1/Makefile
--- support-R6-2-1/Makefile	2021-06-12 02:48:59.000000000 +0800
+++ support-R6-2-1/Makefile	2026-01-18 10:21:23.007001174 +0800
@@ -50,7 +50,7 @@
   RELEASE_FILES += $(wildcard $($(1))/configure/RELEASE_PRODS.local.$(EPICS_HOST_ARCH))
   RELEASE_FILES += $(wildcard $($(1))/configure/RELEASE.local)
   RELEASE_FILES += $(wildcard $($(1))/configure/RELEASE.local.$(EPICS_HOST_ARCH))
-  $(eval $$($(1))_DEPEND_DIRS := $(shell $(GET_DEPENDS) $($(1)) $(1) "$(MODULE_LIST)"))
+  $(eval $$($(1))_DEPEND_DIRS := $(shell $(PERL) $(GET_DEPENDS) $($(1)) $(1) "$(MODULE_LIST)"))
 endef
 
 
@@ -59,7 +59,7 @@
 
 
 # Get all defined values in configure/RELEASE
-RELEASE_LIST = $(shell $(call FIND_TOOL,convertRelease.pl) releaseTops )
+RELEASE_LIST = $(shell $(PERL) $(call FIND_TOOL,convertRelease.pl) releaseTops )
 
 # Filter out the module definitions that point to submodules
 $(foreach mod, $(RELEASE_LIST), $(eval $(call FILTER_TOP_LEVEL,$(mod)) ))
