diff -ur ca-gateway-2.1.3/configure/CONFIG_SITE ca-gateway-2.1.3/configure/CONFIG_SITE
--- ca-gateway-2.1.3/configure/CONFIG_SITE	2021-12-15 21:07:30.000000000 +0800
+++ ca-gateway-2.1.3/configure/CONFIG_SITE	2024-10-26 23:46:10.668836724 +0800
@@ -13,7 +13,7 @@
 #######################################################################
 
 # Use Perl compatible regular expressions (PCRE) instead of basic regex
-#USE_PCRE=YES
+USE_PCRE=YES
 
 # For PCRE on Linux
 #    requires package pcre-devel (or libpcre3-dev) for compiling
diff -ur ca-gateway-2.1.3/src/gateResources.h ca-gateway-2.1.3/src/gateResources.h
--- ca-gateway-2.1.3/src/gateResources.h	2021-12-15 21:07:30.000000000 +0800
+++ ca-gateway-2.1.3/src/gateResources.h	2024-10-26 22:02:35.699365380 +0800
@@ -15,8 +15,8 @@
 #define __GATE_RESOURCES_H
 
 #define GATE_LOG_FILE       "gateway.log"
-#define GATE_SCRIPT_FILE    "gateway.killer"
-#define GATE_RESTART_FILE   "gateway.restart"
+#define GATE_SCRIPT_FILE    ""
+#define GATE_RESTART_FILE   ""
 #define GATE_PV_LIST_FILE   "gateway.pvlist"
 #define GATE_PV_ACCESS_FILE "gateway.access"
 #define GATE_COMMAND_FILE   "gateway.command"
