diff -ur asyn-R4-44-2/asyn/drvAsynUSBTMC/drvAsynUSBTMC.c asyn-R4-44-2/asyn/drvAsynUSBTMC/drvAsynUSBTMC.c
--- asyn-R4-44-2/asyn/drvAsynUSBTMC/drvAsynUSBTMC.c	2023-03-29 02:09:50.000000000 +0800
+++ asyn-R4-44-2/asyn/drvAsynUSBTMC/drvAsynUSBTMC.c	2026-03-25 22:16:36.421462010 +0800
@@ -344,7 +344,7 @@
     if (details >= 100) {
         int l = details % 100;
         fprintf(fp, "==== Set libusb debug level %d ====\n", l);
-        libusb_set_option(pdpvt->usb, LIBUSB_OPTION_LOG_LEVEL, l);
+        libusb_set_debug(pdpvt->usb, l);
     }
 }
 
