diff -Nur install_image/pi/misc/picam-tmp.conf install_image/pi/misc/picam-tmp.conf
--- install_image/pi/misc/picam-tmp.conf	2022-02-02 03:43:56.000000000 +0800
+++ install_image/pi/misc/picam-tmp.conf	2022-09-14 00:15:57.463371258 +0800
@@ -1 +1 @@
-d /var/run/pits 0777 root root
+d /var/run/pits 0775 root video
diff -Nur install_image/picam.rules install_image/picam.rules
--- install_image/picam.rules	1970-01-01 08:00:00.000000000 +0800
+++ install_image/picam.rules	2022-09-13 00:39:41.737426787 +0800
@@ -0,0 +1,1 @@
+SUBSYSTEM=="usb", ATTR{idVendor}=="0647", GROUP="video", MODE="0660"
